src/indicators/money_flow_index.rs

```diff
@@ -0,0 +1,235 @@
+use std::collections::VecDeque;
+use std::fmt;
+
+use crate::errors::*;
+use crate::{Close, High, Low, Next, Reset, Volume};
+
+/// Money Flow Index (MFI).
+///
+/// The MFI is an volume and price based oscillator which gives moneyflow over n periods.
+/// MFI is used to measure buying and selling pressure.
+/// MFI is also known as volume-weighted RSI.
+///
+/// # Formula
+///
+/// Typical Price(TP) = (High + Low + Close)/3
+/// Raw Money Flow(RMF) = Typical Price x Volume
+/// RMF is positive when currnt TP is greater that previous period TP and
+/// negative when current TP is less than preivous TP
+/// Positive money flow (PMF)- calculated by adding the money flow of all the days RMF is positive
+/// Negative money flow (PMF)- calculated by adding the money flow of all the days RMF is negative
+/// Money Flow Index(MFI) = 100 * (PMF / (PMF + NMF) )
+///
+/// Where:
+///
+/// _tp_ - typical price
+/// _rmf_ - raw MF
+/// _pmf_ - positive MF
+/// _nmf_ - negative MF
+///
+/// # Parameters
+///
+/// * _n__ - number of periods, integer greater than 0
+///
+/// # Example
+///
+/// ```
+/// use ta::indicators::MoneyFlowIndex;
+/// use ta::{Next, DataItem};
+///
+/// let mut mfi = MoneyFlowIndex::new(3).unwrap();
+/// let di = DataItem::builder()
+///             .high(3.0)
+///             .low(1.0)
+///             .close(2.0)
+///             .open(1.5)
+///             .volume(1000.0)
+///             .build().unwrap();
+/// mfi.next(&di);
+///
+/// ```
+/// # Links
+/// ![Moeny Flow Index, Wikipedia](https://en.wikipedia.org/wiki/Money_flow_index)
+/// ![Money Flow Index, stockcharts] (https://stockcharts.com/school/doku.php?id=chart_school:technical_indicators:money_flow_index_mfi)
+
+#[derive(Debug, Clone)]
+pub struct MoneyFlowIndex {
+    n: u32,
+    mf_vec: VecDeque<f64>,
+    prev_tp: f64,
+}
+
+impl MoneyFlowIndex {
+    pub fn new(n: u32) -> Result<Self> {
+        match n {
+            0 => Err(Error::from_kind(ErrorKind::InvalidParameter)),
+            _ => {
+                let indicator = Self {
+                    n: n,
+                    mf_vec: VecDeque::with_capacity(n as usize + 1),
+                    prev_tp: 0.0,
+                };
+                Ok(indicator)
+            }
+        }
+    }
+}
+
+impl Next<f64> for MoneyFlowIndex {
+    type Output = f64;
+
+    fn next(&mut self, mf: f64) -> f64 {
+        if (self.mf_vec.len() as u32) < self.n {
+            self.mf_vec.push_back(mf);
+        } else {
+            self.mf_vec.pop_front();
+            self.mf_vec.push_back(mf);
+        }
+
+        if self.mf_vec.len() as u32 == self.n {
```

Comment:
It looks like this `if/else` statement is redundant.

This `if` will be only executed only when the previous `else` is executed (line 84).

Reply:
changed it in new commit